CoffeeAuth/UserPage.xaml.cs

```diff
@@ -0,0 +1,294 @@
+#endif
+
+        }
+
+        int numTicks;
+        TextBlock countdown;
+
+        private void Timer_Tick(object sender, object e)
+        {
+            DispatcherTimer timer = (DispatcherTimer)sender;
+            numTicks--;
+            if (numTicks == 0)
+            {
+            // Turn off grinder
+#if !HARDWARE
+                App.arduino.digitalWrite(13, PinState.LOW);
+#endif
+                Frame.Navigate(typeof(MainPage));
+                countdown_dialog.Hide();
+                timer.Stop();
+            }
+            else
+            {
+                //update timer UI 
+                countdown.Text = numTicks.ToString();
+                if (numTicks <= 15)
+                    countdown.Foreground = new SolidColorBrush(Colors.Orange);
+                if (numTicks <= 5)
+                    countdown.Foreground = new SolidColorBrush(Colors.Red);
+            }
+        }
+
+
+        private async void Countdown()
+        {
+            TextBlock text = new TextBlock();
+            text.HorizontalAlignment = HorizontalAlignment.Center;
+            text.Text = "You have 30 seconds to grind your coffee";
+
+            countdown = new TextBlock();
+            countdown.Text = "30";
+            countdown.FontSize = 48;
+            countdown.HorizontalAlignment = HorizontalAlignment.Center;
+            countdown.Padding = new Thickness(0, 10, 0, 0);
+
+            StackPanel stack = new StackPanel();
+            stack.Margin = new Thickness(0, 40, 0, 0);
+            stack.Children.Add(text);
+            stack.Children.Add(countdown);
+
+            countdown_dialog = new ContentDialog();
+            countdown_dialog.Content = stack;
+            SolidColorBrush color = new SolidColorBrush(Colors.Black);
+            countdown_dialog.Background = color;
+            await countdown_dialog.ShowAsync();
+        }
+
+        private void bagButton_Click(object sender, RoutedEventArgs e)
+        {
+            showToast("Espresso Beans Deposited", "$14 credited to your account.", "Everyone appreciates your efforts");
+            m_user.Balance += 14;
+            m_user.NumBags++;
+            userBalance.Text = m_user.Balance.ToString();
+            DrinkerDatabase.Instance.UpdateUser(m_user);
+        }
+
+        private void milkButton_Click(object sender, RoutedEventArgs e)
+        {
+            showToast("Milk Jug Deposited", "$6 credited to your account.", "Everyone appreciates your efforts");
+            m_user.Balance += 6;
+            m_user.NumMilks++;
+            userBalance.Text = m_user.Balance.ToString();
+            DrinkerDatabase.Instance.UpdateUser(m_user);
+        }
+
+        private void AppBarButton_Click(object sender, RoutedEventArgs e)
+        {
+            Frame.Navigate(typeof(MainPage));
+        }
+
+        private void showToast(string heading, string body, string body2)
+
+        {
+            var builder = new StringBuilder();
```

Comment:
This looks hinky. Can you load the XAML instead of constructing it from a string?